packages/trace-viewer/src/ui/networkResourceDetails.tsx

```diff
@@ -128,11 +132,33 @@ const BodyTab: React.FunctionComponent<{
 
   return <div className='network-request-details-tab'>
     {!resource.response.content._sha1 && <div>Response body is not available for this request.</div>}
+    {responseBody && responseBody.font && <FontPreview font={responseBody.font} />}
     {responseBody && responseBody.dataUrl && <img draggable='false' src={responseBody.dataUrl} />}
     {responseBody && responseBody.text && <CodeMirrorWrapper text={responseBody.text} mimeType={responseBody.mimeType} readOnly lineNumbers={true}/>}
   </div>;
 };
 
+const FontPreview: React.FunctionComponent<{
+  font: BinaryData;
+}> = ({ font }) => {
+  React.useEffect(() => {
+    const fontFace = new FontFace('font-preview', font);
```

Comment:
Let's try/catch this to handle invalid syntax errors during font loading.

Reply:
According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/FontFace/FontFace#exceptions) and to the [spec](https://drafts.csswg.org/css-font-loading/#font-face-constructor), the constructor will throw a `DOMException`. Makes me think we should still have try/catch. I haven't tried it out myself though - what makes you say it doesn't throw any errors?

Reply:
From what I have checked `FontFace` rejects an internal promise (`loaded`) with `DOMException`.
Nothing seems to be thrown from the constructor.

It's quite confusing so I could be wrong. So let's add try/catch to make sure there are no unhandled errors.

<img width="820" alt="Screenshot 2024-08-19 at 15 39 38" src="https://github.com/user-attachments/assets/41eef6dc-a869-43eb-ac2a-c9c31eea9bd2">
<img width="757" alt="Screenshot 2024-08-19 at 15 43 18" src="https://github.com/user-attachments/assets/f90a3d15-eab0-4d1d-ba2c-8f9ffbe0f3f2">

Reply:
True, but then a couple lines down it talks about "terminating the algorithm". I could imagine that some implementations interpret that as "throw an exception".
<img width="767" alt="Screenshot 2024-08-19 at 15 57 06" src="https://github.com/user-attachments/assets/7232f119-d695-4196-8561-059e3dc350d0">

Anyways, having try/catch is good! Thank you.